### PR TITLE
Add Sentry to the project

### LINKED ISSUE #105: Add Sentry for error reporting
https://docs.sentry.io/clients/java/modules/android/?platform=javascript

### PR DESCRIPTION
**Pull request by devinreams**
Fixes #105 

- [x] add Sentry
- [x] test a failure/exception and make sure it records in Sentry
- [x] add to ProGuard and upload the rules to Sentry
- [x] set DSN as variable

### CODE REVIEW

**Comment by devinreams** [comment]
OK I have the test Sentry crashes reporting:

image

https://sentry.prod.mozaws.net/operations/lockbox-android/dashboard/

There is more we can do if we want (set the "environment") but this feels like a good place to start...

**Comment by sashei** [comment]
✨emails✨

**Comment by devinreams** [comment]
OK I added proguard rules and was going down a path of having it uploading rules but realized we may not actually obfuscate our code so don't need it. I'm leaving it out for now.

This is ready for review! ✅

File: app/src/test/java/mozilla/lockbox/store/DataStoreTest.kt [hunk 1]
```diff
@@ -58,7 +58,7 @@ class DataStoreTest : DisposingTest() {
         subject.unlock().subscribe(waiter)
         waiter.assertComplete()
 
-        waiter = createTestObserver<Unit>()
+        waiter = createTestObserver()
```
Comment (sashei) [inline app/src/test/java/mozilla/lockbox/store/DataStoreTest.kt]: nice.

File: app/src/main/java/mozilla/lockbox/LockboxApplication.kt [hunk 2]
```diff
@@ -6,15 +6,24 @@
...
 import android.app.Application
+import io.sentry.Sentry
+import io.sentry.android.AndroidSentryClientFactory
 import mozilla.components.support.base.log.Log
 import mozilla.components.support.base.log.logger.Logger
 import mozilla.components.support.base.log.sink.AndroidLogSink
 
 val log: Logger = Logger("Lockbox")
 class LockboxApplication : Application() {
+    @SuppressLint("AuthLeak")
```
Comment (sashei) [inline app/src/main/java/mozilla/lockbox/LockboxApplication.kt]: is this actually OK?

File: app/lint-baseline.xml [hunk 3]
```diff
@@ -1,1090 +1,1112 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<!--
```
Comment (sashei) [inline app/lint-baseline.xml]: 😬 everything about this change is a bummer

File: app/src/main/java/mozilla/lockbox/LockboxApplication.kt (hunk 2 above)
Comment (devinreams) [reply app/src/main/java/mozilla/lockbox/LockboxApplication.kt]: Eh.. no, it's not great. I will see about turning it into an environment variable.

File: app/lint-baseline.xml (hunk 3 above)
Comment (devinreams) [reply app/lint-baseline.xml]: I'll start chipping away at the baseline next so we see far less of this nonsense, I promise 😁

File: app/src/test/java/mozilla/lockbox/store/DataStoreTest.kt (hunk 1 above)
Comment (devinreams) [reply app/src/test/java/mozilla/lockbox/store/DataStoreTest.kt]: >nice.

All credit goes to the Android linter ❇️

**Pull request merged**